src/cpp/session/modules/SessionReticulate.R

```diff
@@ -79,6 +79,24 @@
    return(TRUE)
 })
 
+.rs.addFunction("reticulate.initialize", function()
+{
+   # try to default to the tkAgg backend (note that we'll
+   # force a vanilla 'agg' backend when required otherwise)
+   engine <- tolower(Sys.getenv("MPLENGINE"))
+   if (engine %in% c("", "qt5agg"))
+      Sys.setenv(MPLENGINE = "tkAgg")
+   
+   # set up hooks for overriding 'show()' in matplotlib
+   setHook("reticulate::matplotlib.pyplot::load", .rs.reticulate.matplotlib.pyplot.loadHook)
+})
+
+.rs.addFunction("reticulate.matplotlib.pyplot.loadHook", function(plt)
+{
+   .rs.setVar("reticulate.matplotlib.show", plt$show)
+   plt$show <- .rs.reticulate.matplotlib.showHook
```

Comment:
I wonder if we should have an opt-out solution, in case users don't want our hook? This would mean that attempts to show plots could show up in external graphics devices, rather than e.g. the RStudio Plots pane.

Reply:
Seems like an edge case so up to you, really. Maybe as an `option` if we need to?